packages/task/src/browser/quick-open-task.ts

```diff
@@ -167,27 +175,80 @@ export class QuickOpenTask implements QuickOpenModel, QuickOpenHandler {
         const configuredTasks = await this.taskService.getConfiguredTasks();
         const providedTasks = await this.taskService.getProvidedTasks();
 
-        if (!configuredTasks.length && !providedTasks.length) {
+        // check if tasks.json exists. If not, display "Create tasks.json file from template"
+        // If tasks.json exists and empty, display 'Open tasks.json file'
+        let isFirstGroup = true;
+        const { filteredConfiguredTasks, filteredProvidedTasks } = this.getFilteredTasks([], configuredTasks, providedTasks);
+        const groupedTasks = this.getGroupedTasksByWorkspaceFolder([...filteredConfiguredTasks, ...filteredProvidedTasks]);
+        if (groupedTasks.has(undefined)) {
+            const configs = groupedTasks.get(undefined)!;
+            this.items.push(
+                ...configs.map(taskConfig => {
+                    const item = new TaskConfigureQuickOpenItem(
+                        taskConfig,
+                        this.taskService,
+                        this.taskNameResolver,
+                        this.workspaceService,
+                        isMulti,
+                        { showBorder: false }
+                    );
+                    item['taskDefinitionRegistry'] = this.taskDefinitionRegistry;
+                    return item;
+                })
+            );
+            isFirstGroup = false;
+        }
+
+        const rootUris = (await this.workspaceService.roots).map(rootStat => rootStat.uri);
+        for (const rootFolder of rootUris) {
+            const uri = new URI(rootFolder).withScheme('file');
+            const folderName = uri.displayName;
+            if (groupedTasks.has(uri.toString())) {
+                const configs = groupedTasks.get(uri.toString())!;
+                this.items.push(
+                    ...configs.map((taskConfig, index) => {
+                        const item = new TaskConfigureQuickOpenItem(
+                            taskConfig,
+                            this.taskService,
+                            this.taskNameResolver,
+                            this.workspaceService,
+                            isMulti,
+                            {
+                                groupLabel: index === 0 && isMulti ? folderName : '',
+                                showBorder: !isFirstGroup && index === 0
+                            }
+                        );
+                        item['taskDefinitionRegistry'] = this.taskDefinitionRegistry;
+                        return item;
+                    })
+                );
+            } else {
+                const existTheiaTaskConfigFile = await this.fileSystem.exists(uri.resolve('.theia').resolve('tasks.json').toString());
```

Comment:
Only `PreferenceConfiguration` can know which folders are there. Clients should not hardcode them, please refactor it. Also use `PreferenceService` to check existence of tasks.json, don't reimplement it.